editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/features/misc/ReconnectNodesFeature.java

```diff
@@ -57,14 +53,41 @@ public void postReconnect(IReconnectionContext context) {
 		AbstractCamelModelElement oldTarget = getNode(context.getOldAnchor());
 		AbstractCamelModelElement newTarget = getNode(context.getNewAnchor());
 		
-		if (source.getOutputElement().equals(oldTarget)) {
+		if (oldTarget.equals(source.getOutputElement())) {
 			source.setOutputElement(null);
 			oldTarget.setInputElement(null);
 		}
-		if (source.getInputElement().equals(oldTarget)) {
+		if (oldTarget.equals(source.getInputElement())) {
 			source.setInputElement(null);
 			oldTarget.setOutputElement(null);
 		}
 		CamelElementConnection f = new CamelElementConnection(source, newTarget);
-	}	
+		DiagramOperations.layoutDiagram((CamelDesignEditor)getDiagramBehavior().getDiagramContainer());
+	}
+	
+	/**
+	 * Returns the EClass belonging to the anchor, or null if not available.
+	 */
+	private AbstractCamelModelElement getNode(Anchor anchor) {
+		if (anchor != null) {
+			Object obj = getBusinessObjectForPictogramElement(anchor
+					.getParent());
+			if (obj instanceof AbstractCamelModelElement) {
+				return (AbstractCamelModelElement) obj;
+			}
+		}
+		return null;
+	}
+	
+	private boolean isReconnectValid(IReconnectionContext context) {
+		AbstractCamelModelElement source = getNode(context.getConnection().getStart());
+		AbstractCamelModelElement newTarget = getNode(context.getNewAnchor());
+
+		return 	source != null && 
```

Comment:
miss the case of trying to reconnect the source to the first element, even if it is not working after, we don't have the icon which give feedback for the forbidden action

![image](https://cloud.githubusercontent.com/assets/1105127/17691910/73c690ac-6397-11e6-9cb1-682a9555cceb.png)